types/react-dom/experimental.d.ts

```diff
@@ -50,4 +50,23 @@ declare module '.' {
     type FormStatus = FormStatusPending | FormStatusNotPending;
 
     function experimental_useFormStatus(): FormStatus;
+
+    function experimental_useFormState<State>(
+        action: (state: State, formData: FormData) => Promise<State>,
```

Comment:
I played a bit with this already and if I understand it correctly, I think that the payload (formData in this case) should be generic as well? In React repo it seems to be the case with Flow typings

Reply:
I'll bring this up but I don't think it really is.

Reply:
Hm seems like it is. I thought it was meant as a form action but it's really just a generic dispatch.

Reply:
> Hm seems like it is. I thought it was meant as a form action but it's really just a generic dispatch.

Yeah, it worked locally with dispatching it manually with random payload. Thanks for checking it out 😃